src/platform/openiotsdk/KeyValueStoreManagerImpl.cpp

```diff
@@ -35,22 +35,47 @@ namespace chip {
 namespace DeviceLayer {
 namespace PersistedStorage {
 
-static const char prefix[] = "/kv/chip-kvs-";
+static const char prefix[] = "chip-kvs-";
 
 class KeyBuilder
 {
 public:
     KeyBuilder(const char * key)
     {
-        auto ret = snprintf(buffer, sizeof(buffer), "%s%s", prefix, key);
-        valid    = (ret > 0) && ((size_t) ret <= (sizeof(buffer) - 1));
+        auto ret = snprintf(buffer, sizeof(buffer), "%s", prefix);
+        if ((ret > 0) && ((size_t) ret <= (sizeof(buffer) - 1)))
+        {
+            // Check sign by sign if key contians illegal characters
+            // Each illegal character will be replaced by '!' + capital encoded letter value
+            char * out             = buffer + strlen(prefix);
+            size_t buffer_capacity = sizeof(buffer) - strlen(prefix);
+            size_t key_len         = 0;
+            while ((key_len < buffer_capacity) && *key)
+            {
+                if (strchr(illegalCharacters, *key))
```

Comment:
maybe capture the result here so it can be reused
```suggestion
                if (illegal_ptr = strchr(illegalCharacters, *key))
```